### ISSUE TITLE
Auto Upload does not work.

### ISSUE THREAD

**Issue by CrippinNRW** [nextcloud/android#2249]
### Actual behaviour
Auto Upload does not work.
Manual Upload works. But if I take a Picture I get a Message with connection Problems

 

### Environment data
Android version: 7.0

Device model: 

Huawei Mate 10 Lite RNE-L21

Stock or customized system:

stock

Nextcloud app version:

3.0.2

Nextcloud server version:

13

**Comment by mario** [comment]
Manual upload is pretty much the same as auto upload. So dunno where that's come from :( Only guess is you have Huawei and you haven't enabled background tasks for Nextcloud app?

**Comment by xXSTrikeXx** [comment]
I have this problem not so often, but at least on one file. I'll keep an eye on it ;)

**Comment by pixiandreas** [comment]
Hello
Same problem here
Samsung S7 with nextcloud 3.0.2

Before the app update 2018-02-28 everything works as it should.
But  the manual update and Web works but not the autoupload 
The error message is "Could no complete operation. Server unavailable"

I have try to restart the server and the app but still the same error.

Andreas H

**Comment by jaark** [comment]
I am seeing this exact same problem with a Galaxy S6.

The uploads still fail if you manually click to retry but a 'normal' manual upload works fine.

**Comment by jaark** [comment]
I've got what looks like a relevant snippet from the client log file (nothing seems to happen on the server though I could be missing something)

> 
> 
> 2018/03/01 09:18:58
> UploadsStorageManager : Updating /storage/emulated/0/DCIM/Camera/20180301_090234.jpg with status=UPLOAD_IN_PROGRESS
> 
> 2018/03/01 09:18:58
> UploadsStorageManager : updateUpload returns with: 1 for file: /storage/emulated/0/DCIM/Camera/20180301_090234.jpg
> 
> 2018/03/01 09:18:58
> UploadsStorageManager : notifyObserversNow
> 
> 2018/03/01 09:18:58
> FileUploader : Error uploading Exception : [com.owncloud.android.operations.UploadFileOperation.run(UploadFileOperation.java:374), com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:145), com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:149), com.owncloud.android.operations.common.SyncOperation.execute(SyncOperation.java:93), com.owncloud.android.files.services.FileUploader.uploadFile(FileUploader.java:1090), com.owncloud.android.files.services.FileUploader$ServiceHandler.handleMessage(FileUploader.java:1023), android.os.Handler.dispatchMessage(Handler.java:102), android.os.Looper.loop(Looper.java:154), android.os.HandlerThread.run(HandlerThread.java:61)]
> 
> 2018/03/01 09:18:58
> UploadsStorageManager : updateDatabaseUploadResult uploadResult: com.owncloud.android.lib.common.operations.RemoteOperationResult@ad925c2 upload: com.owncloud.android.operations.UploadFileOperation@e025fd3
> 
> 2018/03/01 09:18:58
> UploadsStorageManager : Updating /storage/emulated/0/DCIM/Camera/20180301_090234.jpg with status:UPLOAD_FAILED and result:NETWORK_CONNECTION (old:/storage/emulated/0/DCIM/Camera/20180301_090234.jpg status:UPLOAD_IN_PROGRESS result:-1)]
Does this help any?

**Comment by jospoortvliet** [comment]
@mario my galaxy S7 also gives 'server unavailable' and that is using cloud.nextcloud.com for testing ;-) ping me if you need me to test/check anything.

**Comment by rikmeijer** [comment]
Having the same problem, although nothing relevant appears in my log files

**Comment by digorgonzola** [comment]
Same issue here on a MotoG but the following seemed to fix it.

Uploaded two files manually: worked a treat

Went back to the upload queue and retried failed uploads. New message "waiting to exit powersave mode" (just happened to have pretty low battery).

Plugged in charger and hey presto it worked! 

Took some more photos just to make sure it wasn't a fluke and it did work.

**Comment by jaark** [comment]
The message for me has always been "Failed/pending restart".

Plugging in the power cable made no difference.

**Comment by mario** [comment]
So auto upload has two jobs:
- schedule upload
- auto-retry

It looks like that parts are working.

Then it performs regular upload, so did you all try (including @jospoortvliet) if regular upload is working? Maybe related to agent changes @tobiasKaminsky ?

**Comment by jaark** [comment]
Yes, regular file upload works fine.

**Comment by rikmeijer** [comment]
@mario Yes, regular uploads work

**Comment by FoXFTW** [comment]
Having the same problem.  
Device: Galaxy S8  
Version: 3.0.2 

Manual Upload works fine

**Comment by wvxx** [comment]
Same issue since yesterday's update.
Device: Xiaomi Mi 5
Version: 3.0.2

Tested as well on Huawei Mediapad x2  - same issue
Manual upload works in both cases.

**Comment by tobiasKaminsky** [comment]
Is manually resuming a broken, scheduled by auto upload, one working?
(I tried it with 3.0.2, and NC13 and it is working).

**Comment by jaark** [comment]
No, manually retrying the upload fails.

**Comment by wvxx** [comment]
In my case it doesn't even look like it's retrying anything. Progress bar pops up for literally a few miliseconds, then the same error comes back.

**Comment by jaark** [comment]
Mine seems just like the above, I do not notice anything in the Apache logs when retrying the upload either..

**Comment by xraMsamohT** [comment]
Same here with Android 6 device with 3.0.2 client.
I tried the **Force new scan** (German: Neuen Scan erzwingen) menu entry within Uploads and it seems that Auto Upload of new files (Screenshots) works now.
I'll keep an eye on it.

**Comment by wvxx** [comment]
Force rescan does not work for me, sadly.

**Comment by HardcoreWR** [comment]
Unfortunately both my wife and me have the exact same issue.
Manual upload works, but auto upload schedules the files to be upload but all fail with 'Connection error'

My phone: Galaxy S8
Nextcloud app version: 3.0.2

Wife's phone: Galaxy A5 (2017)
Nextcloud app version: 3.0.2

**Comment by Layiap** [comment]
Same problem here.
Galaxy S8
Android 8.0.0
Nextcloud app version: 3.0.2
Server v 13.0.0

**Comment by lukas2511** [comment]
Same here.
OnePlus 5T
Android 8.0.0
Nextcloud app version: 3.0.2
Server v 13.0.0

**Comment by cmot-weasel** [comment]
Same here on two different Nextcloud server instances:

Personal setup:
* Nextcloud Server v13.0.0 (just upgraded, same effect on 12.0.4, 12.0.5)
* 2x Huawei P10 (both Android 7.0)
* App version 3.0.2

Work Setup:
* Nextcloud server v13.0.0 
* multiple Samsung Xcover 3 (Android 5.1.1)
* App version 3.0.2

Summary: 
* Manual "upload file" works fine
* Auto-upload fails with "Connection Error", manual restart fails
* No logs in Nextcloud app on any device (in expert mode)
* No logs on either server at all (web server or nextcloud.log)
* No affect by removing/readding accounts on app, by clearing app data or reinstalling app.

**Comment by rikmeijer** [comment]
Just added a brand new account to my nextcloud android client, problem persists for auto uploads under that account

**Comment by lukas2511** [comment]
@mario Could this be the issue? https://github.com/nextcloud/android/commit/38dc41ad3bd3ffe3bdd4746e75dce2a1145209c8

The commit message indicates that the code should have been removed, but instead it has been re-activated. And it's exactly at the line mentioned in @jaark's log snippet.

**Comment by mario** [comment]
@tobiasKaminsky any idea about those lines? I think they were just temporarily commented out before, right?

@lukas2511 same code is used for regular upload so if that works...

**Comment by AlexanderKgr** [comment]
I had the same problem. I thought that the only things they have changed from yesterday was the app update and....the month. I manually added the folder 03 under the 2018 folder and the auto upload finished without any problem. I think that the problem is the new app and the change of the month

**Comment by mario** [comment]
Well then the grant folder thing could indeed be at fault. But is it needed
for e2e?

**Comment by xraMsamohT** [comment]
> I manually added the folder 03 under the 2018 folder and the auto upload finished without any problem.

Exactly the same here. That's it!

**Comment by jaark** [comment]
Hmm, after reading @xraMsamohT 's reply, I went to check mine and though I don't have the monthly folder change set up, I had noticed that the upload folder has changed from /InstantUpload/ to /InstantUpload/Camera/

This is a different default behaviour from previous versions and, of course I do not have a Camera folder under InstantUpload. Changing this to upload to InstantUpload makes automatic uploads for new pictures work again.

So, as far as I can see we have three separate issues..
* Default upload location of the latest version of the app does not match previous versions (nor does it match the directory structure created for new NC users)
* Uploads fail when the target directory does not exist (not unreasonable) and the error message returned is totally misleading.
* Monthly folders are not automatically created by the client when that functionality is selected.

**Comment by wvxx** [comment]
> Uploads fail when the target directory does not exist (not unreasonable) and the error message returned is totally misleading.

This does not seem to be the only case. I tried autouploading a photo from another device which didn't have latest version installed. It worked well and the folder was created, yet retrying failed upload on my phone (or trying a new one) is still impossible.

**Comment by rikmeijer** [comment]
Too bad, manually creating a 03 folder did not solve it.

**Comment by mario** [comment]
Let's see if https://github.com/nextcloud/android/pull/2265 fixes it :)

**Comment by rikmeijer** [comment]
I'd be willing to test if I knew how 😊

**Comment by xraMsamohT** [comment]
2nd test with Camera folder. After manually creating the 03 folder the restart of uploading works. Afterwards the Auto Upload works as aspected.

**Comment by Memphizzz** [comment]
I can also confirm this. After creating a folder called "03" in my AutoUploads folder using the web interface, then going back to the app and see that folder appear, AutoUploads work. I guess its just that the app "forgets" to create the new directory.

**Comment by digorgonzola** [comment]
I also had to manually create the new folder for this month "03"On 2 Mar. 2018 10:41, MemphiZ <notifications@github.com> wrote:I can also confirm this. After creating a folder called "03" in my AutoUploads folder using the web interface, then going back to the app and see that folder appear, AutoUploads work. I guess its just that the app "forgets" to create the new directory.

**Comment by m3lloa** [comment]
Creating the month folder didn't fix it for me.

NC v. 13.0.0
FreeBSD 11.1 VM on a FreeBSD 11.1 Host
Apache version 2.4.29
MySQL version 10.1.31
PHP Version: 7.1.13
Android Client: 3.0.2
LG G5 Android 7.0

**Comment by FunkyLoveCow** [comment]
I ran into this problem running 3.0.2 on a Pixel 2 on 2018-03-01 as well. Normally, a new directory would be automatically created at the start of each month, but for some reason it was not during the auto-upload process this month. 

I have no relevant logs on the failure in either the client or the server. Only the message "Could not complete operation. Server unavailable" from the client when it attempts to auto-upload or retry after a failure.

I resolved this by manually creating the `03` directory that was expected.

**Comment by m3lloa** [comment]
Not on my setup. Created 03 folder and issue remains. It says could not complete operation, server unavailable.
When you check the upload on the ap, the status is connection error.
________________________________

**Comment by bryantech** [comment]
dev app 20180208 and 3.02 from Google. ZTE 981

No auto upload or manaul upload. I have restarted Nextcloud 3 times.

**Comment by augin** [comment]
+1

**Comment by hal3993** [comment]
I had the same issue since 1st March. Here is my workaround (like some others before me also wrote):

- Use the web interface and create a new folder named "03" in your AutoUpload folder.
- Use the Android client, browse to your auto upload folder and refresh. Make sure you can see the empty folder "03".
- Use the Android client, go to the "Uploads" section and tap on the failed uploads.
- Now it works for me. (It didn't before the refresh on the Android client.)

**Comment by augin** [comment]
It's work for me. Thanks. Xiaomi Mi6, Xiaomi Mi Note3

**Comment by rikmeijer** [comment]
Apparently I created the folder under the wrong directory. Following @hal3993 instructions fixed it fir me! Thanks!

**Comment by jdkruzr** [comment]
Creating the 03 folder worked for me on a OnePlus 3T with a custom ROM. Obviously I didn't have to do this for February.

**Comment by jospoortvliet** [comment]
@hal3993 that worked for me!

**Comment by derseeli** [comment]
After creating several 03 folders it works for every autoupload. 

OnePlus 3T
Android 8.0.0
Nextcloud app version: 3.0.2
Server v 13.0.0

And Huawei Nova 2 (Android 7?)

**Comment by wvxx** [comment]
hal3993's solution works for me as well. Thanks mate!

**Comment by evanrich** [comment]
@hal3993 saves the day

**Comment by guoyunhe** [comment]
Same problem.

**Comment by Frank1967** [comment]
Confirmed on my Samsung S5 as well: creating the "03" upload directory fixes the problem for the Android 3.0.2 version. 

Also confirmed that the problem does not exist in the 3.0.1 version of the client (just checked on a phone with automatic updates disabled).

**Comment by Railsimulatornet** [comment]
same problem on galaxy s8 Oreo

**Comment by timeggleston** [comment]
Same issue and resolution on a Pixel 2 with Android 8.1.0, creating the 03 directory fixes the issue and allows the failed uploads to be retried successfully.

**Comment by dsiminiuk** [comment]
Moto G5 Plus NC v3.0.2
Autouploading

"Upload Failed"
"Could not complete operation, Server unavailable"

But I can browse the store manually.

**Comment by ppoetsma** [comment]
Same auto upload issue here on Android 7 devices with latest Nextcloud app 3.0.2. Devices are Motorola Moto G5S Plus and Samsung Galaxy S7. Creating the 2018 and 03 folders do not help. Browsing the files and downloading them works fine, it's just the auto upload of photo's and probably also videos.

Auto upload works fine on Apple iPhone 5S.

**Comment by Wikinaut** [comment]
Creating the "03" folder "solved" the issue. 

#### Regression!
Now the question remains: why is the "03" not automatically created? This was the case earlier.

**Comment by mario** [comment]
Like @tobiasKaminsky mentioned, we hope to have this fixed in 3.0.3, aka soon.

**Comment by freelock** [comment]
I did try creating the "03" directory in the web interface, but it still failed to work... until I browsed there in Android and refreshed... Once Android "saw" the directory, auto uploading worked again...

**Comment by OlivierS1** [comment]
This was not working for me either but then reading on the above comment I came to a fix on mine:

Creating the 03 folder from within the web interface to my Nextcloud account under my Instant upload folder within the 2018 folder.
This looks like this:

instantupload/camera/2018/03

Trying to upload failed uploads from my phone was still unsuccesful.
However, doing the following did fix the issue:

Within Nextcloud phone app:
1) Exit "Uploads" and return to "Files"
2) do "Refresh account", I did actually go into the "instantupload" folder, then "Camera" then 2018.
Force a refresh.
3) Return to "Uploads" from the 3 bars.
4) Click on the failed uploaded files. They should start uploading.

**Comment by chujionglie** [comment]
I have this problem too. But it's maybe not the fault of App version 3.0.2. Because after I installed a 3.0.0 version, it does not work neither.

**Comment by gqgunhed** [comment]
Nextcloud app 3.0.2 on Sony Xperia Z1 with Android 7.1.2, instantupload with subfolders enabled:
What worked for me is:
- manually create the subfolder "03" within your "2018" folder
- make sure that your nextcloud mobile client shows this newly created folder by browsing/refreshing the sync folders
- if the "03" is visible on the mobile client
- got to "Uploads"
- restart your pending uploads
- should now begin to upload the files into the newly/manually created "03" folder.

You can check which folder needs to be created if you look below the filename in the failed uploads screens.
So it's the subfolder creation part that fails. As other instantupload folders _without subfolders_ just sync fine.

**Comment by leafcutterant** [comment]
For me it "auto"-uploaded instantly only after I went to Uploads and forced a rescan.

Without that, I had to wait ~15 minutes for the sync to happen.

**Comment by caco3** [comment]
I had the same issue.
Android 7.1.2,
Nextcloud App 3.0.2
The fix was to create the folder 03 manually on the server, load the folder on the android app, then retriggered the upload.

**Comment by sebw** [comment]
Same here.

Stopped working on March 2nd.

Moto G5 Plus.
Nextcloud 3.0.2.

**Comment by jospoortvliet** [comment]
So 3.0.3 fixes it? Great, thanks!!! Of course it can only really be tested at the beginning of next month :hankey:

**Comment by AndyScherzinger** [comment]
> Of course it can only really be tested at the beginning of next mont

Nope you just need to activate the monthly folder option for a folder where you didn't had it activated before ;)

**Comment by OlivierS1** [comment]
You can rename the 2018 folder and see if the fix works.

**Comment by xXSTrikeXx** [comment]
sebw tested it and now with latest version 3.0.3 it is working. I close this issue now, feel free to reopen ;)

**Comment by xpilgrim** [comment]
But, now I'm facing a strange login failure of the desktop (linux) client! According to the nextclod (13) server log, the desktop client is unable to login **directly** after updating the android app on my phone.

**Comment by xXSTrikeXx** [comment]
@xpilgrim Open a new Issue in the right repo. You are here at Android ;)

**Comment by m3lloa** [comment]
@xXSTrikeXx - can't make it work on my S5. Will do more tests and open another case if needed. (My LG G5 works fine :( )

**Comment by fozters** [comment]
Yep, fixed issue for me creating the 03 month folder under the camera folder! So it seems that the use subfolders option is unable to create needed folder!

**Comment by fozters** [comment]
Oh this was closed as 3.0.3 should fix it. Didn't notice there was update and didn't notice this thread was closed! Sorry and thanks! :)

**Comment by chefboyct** [comment]
Hi there, 

I am currently on v3.03 and running Oero 8.0 on Sony Xperia XZ G8141 and on the auto upload, it doesnt seem to upload any videos (pictures were ok). But it was working previously (just checked on my cloud and it uploaded some mp4 files on v3.02 I am sure... ? Please advise if there is any workaround on this issue or if this is ongoing... thanks,

**Comment by AndyScherzinger** [comment]
@chefboyct This should work (on any 3.x release) but there should be 2 entries within the auto upload screen for both media types

**Comment by AndyScherzinger** [comment]
@jult did you take pictures etc after you activated the auto upload? Just asking since the auto upload doesn't yet have the feature of uploading already existing files.

**Comment by jult** [comment]
@AndyScherzinger When I do that, take a new photo, and switch on dated folders as mentioned above, the new photo files do appear under Upload, but they fail, under filename it says "Local file not found".
If you need any logs, let me know. My Sammy is rooted etc.

**Comment by OlivierS1** [comment]
@jult I have seen this behavior when the option is set to delete the source file after auto upload.
What I have noticed is that, under certain circumstantces, when transfer starts (over Wifi) but the connection is dropped, NC app would already have deleted the source file, however the auto upload folder would still try to complete the transfer.
I have lost a few pictures in the past because of it.

**Comment by jult** [comment]
@OlivierS1 The 3 pictures I took have indeed been deleted at the source. I had not set it to do so, 'Original file will be ... kept in original folder", I let that as it was. So, what this Auto Upload now does is pretty severe:
**Set a folder to Auto Upload and it will make new photos disappear**! They're not even being uploaded.
If I were you, I'd drop this version release to alpha. This is definitely not a beta release.

**Comment by AndyScherzinger** [comment]
@jult I am sorry to hear that and understand your frustration but on the other hand can't accept such a statement since it is not true. This deletion doesn't happen for the dev's devices and also hasn't been reported by any of the more than 6000 beta tested during the last weeks of the beta phase.


So let's see if we can find the cause for the seen behavior on your installation.
Pinging @mario since he is the auto upload grand master.

**Comment by mario** [comment]
@AndyScherzinger this part of code (aka deleting after upload) hasn't changed in like forever and I have never experienced this nor heard about it :(

**Comment by jult** [comment]
@mario You're presenting a feature (Auto upload) in an Android app in the Playstore as if it is finished, while it isn't functional (for android 6.01 anyway). What caused the files to be deleted in my case could also be related to a 128 GB micro SDcard that is starting to show bad behavior as of late, it's getting slower than usual.
I tested it again; When I take a picture it shows up in the Gallery as normal (stored on that SDcard). I wait a while, switch on the "Use subfolders" preference, a minute later nextcloud app shows the file under Uploads, again with "Local file not found". Poof, it's gone.

Syncing from my nextcloud desktop app works entirely as intended.

**Comment by OlivierS1** [comment]
@mario @AndyScherzinger I suspect the issue could be similar to this post:

https://github.com/nextcloud/android/issues/1445

**Comment by AndyScherzinger** [comment]
Well, #1445 is as stated in the issue set the upload behavior to move the images to the app folder. From what I understood/implly @jult is using the upload behavior "keep in original place" so there shouldn't be any file operation applied to the original file except creating an internal copy for uploading.

**Comment by OlivierS1** [comment]
@AndyScherzinger True. My statement was also to supplement information which are no longer up to date but could have a common root cause.
It is also possible @jult issue is related to another application conflicting with how NC App is handling the files.
It is also conceivable he has more that one account setup on the NC App for which transfer behavior are setup but forgotten.

**Comment by AndyScherzinger** [comment]
@OlivierS1 :+1: true, there are several scenarios possible which could explain the behavior besides the one where it would be a bug.

**Comment by jult** [comment]
@mario I will logcat on my phone a.s.a.p. this weekend and see what's actually going on on my Note 4 (yep, it's a rather 'old' phone, but hey, it still works fine for me). Must work on something else right now.
My issue seems different from others posting here, since I don't see any connectivity warnings/lost connections whatsoever. My nc app is just silently not doing anything, not even after tagging it under Accounts to enable Sync..

**Comment by Cole128** [comment]
Is this fixed now?

**Comment by jult** [comment]
Nope. Uploading barely works at all, using the Android app, not even manually. I'll fire up logging a.s.a.p. Can't Nextcloud use Lsyncd code https://github.com/axkibe/lsyncd ? What strikes me as strange is the way nextcloud doesn't say at what side it will assume files can be deleted. It should offer read only syncs, or use a Master-Slaves principle, but for me it's not even made clear what it does where when I sync a folder, I expected something more configurable. Also, it completely disregards actions outside of nextcloud within server folders, which is kind of strange as well. I have to do a files:scan with console.php from shell everytime something changes, which is super-annoying since I have ~280000 files up there. lsyncd anyone?

**Comment by jult** [comment]
Just for your information; Creating a new non-admin user seems to have fixed the bug for me. So:
Uploading and Auto Upload doesn't work for the server admin user, it does work for a non-admin user.
You may want to document this somewhere.

**Comment by mario** [comment]
This makes no sense @jult. @tobiasKaminsky I've seen this on encrypted stuff - could be related?

**Comment by jult** [comment]
@mario, well, perhaps by installing the new nextcloud account it used some newer code from a newer nextcloud release? I wouldn't know. I'm doing it all on the same android device to same server setup, and this is what made it work for me;
Wiped data/Uninstalled/purged the nextcloud app using Titanium Backup (paid version) on my Samsung N910C. Had to do this because I could not find another way to remove the account without being clear it would NOT take all set-to-sync folders (down) with it, I only wanted to login with a new/different account, not a 'secondary' within the same app. So, I installed the android app anew, logged in using the non-admin user, and lo and behold: It synced without deleting the source, and followed all my config prefs.
Only thing missing is that I think it should sync a dir including what is already there, not just newly created files, and it should have an option to set the sync as read-only. (lsyncd and syncthing anyone?)

**Comment by mario** [comment]
We're going to support syncing existing files eventually. There's only so much of us, so please some patience! :)

**Comment by OlivierS1** [comment]
@Jult, I don't think Admin account is the cause of the issue. My account is admin, not the original admin though and I do not experience particular problems.
I do seem to remember having Non Dev and Dev version of the app installed on the same device  was causing some issues. This was back around NC 1.5 or 2.0.
I think there is a conflict, probably because they were sharing common files. I just don't remember what the issue was, but Autoupload could have been it.

**Comment by Cole128** [comment]
So it seems like maybe the issue is when you use the original account?

**Comment by Cole128** [comment]
it still doesn't work with an admin but not original account

**Comment by jult** [comment]
Did you actually remove the app and its data and cache, before entering the new account? Android is famous for not changing anything about an installed app if you don't do that.. You need to really purge everything of the app, except for your own sync-dirs/content of course.
Also, @OlivierS1 I first installed the Nextcloud Talk app by accident, also added my login to it, only to discover I had picked a totally wrong app (I thought it was built into that Talk app or something, somehow it came up first when I searched on nextcloud in the playstore..). Then installed the (right) Nextcloud app, and a day later removed the Talk app. So no dev app on there yet. But the fact that I first had the Talk app on there may have caused it then? Either way, after purging both and logging in with a new user it now works.

**Comment by tobiasKaminsky** [comment]
There was once a bug on E2E, where the access (and thus also upload) to an encrypted folder did not worked, if the username on login was not the same like the userid (which the admin specified). As most of the admins have an account and login "admin", it seemed that "admin" worked, whereas some other complex usernames did not worked.

This was only related to E2E, so if you do not have this enabled, but still experience an issue with auto upload, it "should" be something different...

**Comment by Cole128** [comment]
@jult so I downloaded a new file to a folder setup to be auto synced, and that file started syncing, but all the files that were in those folders before nextcloud did not sync. Any ideas about that? Also, how do I purge all app data? I can try and do that and then reinstall and see what happens.

**Comment by chefboyct** [comment]
@AndyScherzinger Hi Andy, sorry random question as I cannot seem to find this topic at all under the main discussion and I cannot seem to message you in PRV. I have a quick question as I dust off my old Nokia 1020 (Windows 10 mobile) as I want to use the Nextcloud app to upload pictures to the nextcloud server but I am unsure what Nextcloud windows mobile support or not.. (trial version can only access files and have no options at all to auto upload). Can you help me to understand and verify this ? and if not, can I use owncloud application for Windows 10 mobile to auto upload? Thanks mate!

**Comment by AndyScherzinger** [comment]
@chefboyct don't know really, I just use the Android and iOS apps don't know about the WindMobile capabilities or the oC windows mobile client's features.

**Comment by chefboyct** [comment]
Thanks @AndyScherzinger  I want to know but its fine. I can always manually upload them

**Comment by butterbrot21** [comment]
Same here.
OnePlus 3T
Android 8.0.0
Nextcloud app version: 3.2.0
Server  13.0.4

**Comment by alaunay** [comment]
Same here, Galaxy S9+, app 3.2.1, server 13.0.2. I wonder if there's not something wrong with ipv6.

**Comment by jospoortvliet** [comment]
@makeworld-the-better-one you can do a re-scan of all files.

**Comment by jspearo** [comment]
How can I do a re-scan? 
My instant upload works accidentaly, sometimes some files are uploaded... but usually not. And e.g. for Conversations (jabber) app images are usually uploaded, but when I take screeshot, usually not.
Samsung Galaxy S9+ ,Android version 8.0.0, Nextcloud Android app version 3.2.1, Nextcloud version 12.0.7.

**Comment by Joshndroid** [comment]
I can also confirm that setup with a Moto G5 plus that auto upload of images are not working. highly frustrating.

**Comment by Oxyaxion** [comment]
Hi, 
Same problem for me here. All was working fine before the last Nextcloud Android client update. I use an Honor 9 Lite and Android 8.

Now the automatic sync doesn't want working. I need to push manually all. And sometimes I have some connection problem (I need often to repush my password in the app).

That's bad :(

**Comment by kasumiru** [comment]
I had the same problem. Manual upload works, but auto upload work not. Lg g5, android 7.0, Nextcloud v 12.

**Comment by bjschuit** [comment]
Still an issue. Not always auto upload works. My phone, despite disabling power optimization for Nextcloud, sometimes terminates the app, causing upload failures. Other times it won't initiate. And sometimes it does initiate and starts uploading, but in the end not all of my pictures are uploaded, making this feature completely untrustworthy unfortunately.

**Comment by feutl** [comment]
Using a Galaxy S9 and some pictures end up in the upload queue and some not. I have not figured out which an why, but it worked for some time without problems. started to make problems yesterday.

Even a "force rescan" does not place them in the queue.

I am not having nextcloud background sync activated because it drains 30% of the battery on its own.
Using the lates 13 server version and latest android version from the play store.

**Comment by rwat1** [comment]
@feutl @bjschuit @kasumiru @Oxyaxion @Joshndroid @jspearo @alaunay @butterbrot21 @chefboyct @AndyScherzinger @makeworld-the-better-one @tobiasKaminsky  @OlivierS1 @mario 

Hi guys, I posted a solution here that works for many users. 

https://github.com/nextcloud/android/issues/2834#issuecomment-413856687

TLDR: clearing NC app cache/data  + reinstalling.

**Comment by feutl** [comment]
hmm, but this also means, that the pictures not being uploaded yet are not getting uploaded as well. because we start over again.
This also implies that I do not really know which pictures are missing right now, because it just uploaded some picked them randomly and I have no clue which ones if I do not compare them by hand.

Not really a solution I would say, more a "dirty" workaround.
Especially because it worked at the beginning but for whatever reason stopped partially on the weekend.

**Comment by rwat1** [comment]
@feutl 

> Sth to note, all the steps you described in your post, I did at the very beginning of getting my s9+. Install the app, only the stock gallery app in use and but nextcloud to the "non-standby apps" but still I am now in the situation pictures do not get uploaded any more. I did at the beginning but after some weeks and around 50 pics it stopped working.

Are you me? This is exactly what happened to me too, I applied https://github.com/nextcloud/android/issues/2834#issuecomment-413856687 then few weeks later, same period. I had to apply the fix again. LOL

I think it must be data configuration settings gets corrupted or de-linked whenever there is an Android security update.

**Comment by jult** [comment]
FYI; I've stopped using this app because it just fails precisely when I don't want it to.
Started using app called FolderSync using WebDAV instead with my nextcloud server. Works fine. Even tempted to buy the pro version.

**Comment by feutl** [comment]
@jult
I looked at FolderSync but it is not actively developed since a long time, which the maintainer also explains (https://plus.google.com/+AndersCarlsen/posts/5jVkiakqcFv)
This is not an option for me, if something is not getting developed any more, for whatever reason

@rwat1 
I think I figured it out, for me at least.
I looked at all the devices I have with nextcloud auto upload activated. This are 6 devices all of the work except mine. They are different brands, different android version, Some older some newer, some from SD card some sync from internal storage.
The only thing that popped was *SD card encryption*

Mine was the only one having *SD card encryption* turned on and stores the pictures on the external SD card. I turned that off, and after completing Nextcloud started to resync the pictures I was missing. It did not resync the pics between the problem appeared and I tried stuff and did pictures and turning of SD card encryption, but the ones I was missing from the weekend where Nextcloud just uploaded some but not all.
AND it is also uploading all pics I did since turning it off without problems.

The question is, could it be that SD card encryption also comes into your way?
If yes, try turning it off, I know not very "secure" but so we could just pinpoint the issue ;) because your solution just is not a solution at all.

**Comment by powerriegel** [comment]
Same problem here. Auto Upload does not recognize that I've taken a picture. Huawei Mate 10 Pro. Manual upload works fine.

**Comment by xXSTrikeXx** [comment]
@powerriegel AFAIK the autouloader checks every 15 minutes for new files to prevent battery drain. Please recheck your device behavior with this knowlege ;)

**Comment by Shucach** [comment]
I have same problem. In meizu m2 mini

**Comment by powerriegel** [comment]
@xXSTrikeXx it never uploads anything automatically. I took a picture two days ago and its not uploaded until now.

**Comment by rwat1** [comment]
@powerriegel @Shucach 

This worked on my phone and helped many people too: 

https://github.com/nextcloud/android/issues/2834#issuecomment-413856687

**Comment by Mebus** [comment]
Hallo,

I have got the same problem on a **LG G3** since a few weeks, Android Version 6.0. I can upload files manually, but uploading files automatically doesn't work, even if you try to force the app to retry the upload.

I think the solution, which powerriegel described is only a workaround and doesn't fix the issue. 

Can someone please reopen the ticket?

Mebus

**Comment by rwat1** [comment]
@Mebus 

Have you tried my solution? It fixed my problem and many other people's problem.

https://github.com/nextcloud/android/issues/2834#issuecomment-413856687

**Comment by phi-el** [comment]
Hi,

same Problem on a Sony Xperia XZ2 w/ 3.4.1 where auto upload is also allowed via 4G. Photo automatically uploads when there is no WiFi, but does not automatically starts when there is WiFi.
I'm not able to re-queue failed auto uploads w/ force sync.
The same behavior has been watched on a Samsung J5, but the forced sync uploads the missing photos.

**Comment by IIPoliII** [comment]
Same problem here android 6.0 (custom) on a oneplus 3t I makes an error everytime I try to use it...

**Comment by ChrisMichl** [comment]
Same problem here with my Xiaomi Mi Mix 2

**Comment by rwat1** [comment]
Auto-upload uses GPS to detect changes right? Why would you use that if it drains your battery by leaving GPS on all the time?

**Comment by jospoortvliet** [comment]
Hi, for everyone who has issues in a enterprise setup - be sure to report this through https://portal.nextcloud.com to get priority on this. As there are a LOT of different devices, it is pretty much inevitable that there are some problems left and we have to focus our resources between our roadmap, issues that hit a majority of users and our customers....

**Comment by GnaXi** [comment]
I'm having this issue on a OnePlus 5T running Android 9.
I am however not having the issue on another device, OnePlus  6T running Android 10.

On the OnePlus 5T nothing is happening at all, when enabling the auto upload.
Manual upload works just fine. I've tried re-installing the application several times after removing data and cache. Also restarting the device doesn't make a change.
I've also tried to let it upload without WiFi and disable WiFi, but issue remains exactly the same.

I've also tried enabling beta updates in the Google Play Store.

**Comment by GnaXi** [comment]
I just saw this commit being released not long ago:
https://github.com/nextcloud/android/commit/f477a6873b5b695360c1c91299ab278588011377

Testing it out and so far it seems to have been doing wonders to this issue!
My phone has been sleeping for a bout 30 minutes now and it's just uploading photos to my Nextcloud! :)
Good going @ezaquarii and @tobiasKaminsky ! :D
